File: Backend/dependency/inquiry/src/main/java/com/mhp/coding/challenges/dependency/inquiry/InquiryService.java
package com.mhp.coding.challenges.dependency.inquiry;

import lombok.RequiredArgsConstructor;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.ApplicationEventPublisher;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class InquiryService {

    private static final Logger LOG = LoggerFactory.getLogger(InquiryService.class);

    private final ApplicationEventPublisher applicationEventPublisher;

    public void create(final Inquiry inquiry) {
        LOG.info("User sent inquiry: {}", inquiry);
        applicationEventPublisher.publishEvent(inquiry);
    }

}

File: Backend/dependency-kotlin/inquiry/src/main/kotlin/com/mhp/coding/challenges/dependency/inquiry/InquiryService.kt
package com.mhp.coding.challenges.dependency.inquiry

import mu.KotlinLogging
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.ApplicationEventPublisher
import org.springframework.stereotype.Component

private val logger = KotlinLogging.logger {}

@Component
class InquiryService(
    @Autowired val applicationEventPublisher: ApplicationEventPublisher
) {
    fun create(inquiry: Inquiry) {
        logger.info {
            "User sent inquiry: $inquiry"
        }
        applicationEventPublisher.publishEvent(InquiryEvent(this, inquiry))
    }
}

data class Inquiry(
    var username: String,
    var recipient: String,
    var text: String,
)
